tornado/platform/common.py

```diff
@@ -84,8 +84,10 @@ def consume(self):
                 result = self.reader.recv(1024)
                 if not result:
                     break
-        except (IOError, socket.error):
-            pass
+        except (IOError, socket.error) as e:
+            if e.errno = errno.ECONNRESET:
```

Comment:
That's a syntax error. I'd also reverse this comparison and raise any error that is not equivalent to EWOULDBLOCK; most errors are likely to be persistent.